x-pack/plugins/lens/public/app_plugin/lens_top_nav.tsx

```diff
@@ -381,6 +381,24 @@ export const LensTopNavMenu = ({
     }
   }, [query]);
 
+  useEffect(() => {
+    if (
+      initialContext &&
+      isOfAggregateQueryType(query) &&
+      'dataViewSpec' in initialContext &&
+      allLoaded
+    ) {
+      dispatch(
+        switchAndCleanDatasource({
```

Comment:
When is this triggered? Couldn't figure it out.

Reply:
This is triggered when I navigate with context (from Discover to Lens) and the query is text based.